app/controllers/blogs_controller.rb

```diff
@@ -3,13 +3,20 @@
 class BlogsController < ApplicationController
   skip_before_action :authenticate_user!, only: %i[index show]
 
-  before_action :set_blog, only: %i[show edit update destroy]
+  before_action :set_blog, only: %i[edit update destroy]
 
   def index
     @blogs = Blog.search(params[:term]).published.default_order
   end
 
-  def show; end
+  def show
+    @blog =
+      if current_user.present?
+        Blog.published.or(Blog.where(user_id: current_user.id)).find(params[:id])
```

Comment:
オブジェクトが指定できるときはそうしましょう。

```suggestion
        Blog.published.or(Blog.where(user: current_user)).find(params[:id])
```